solutions/usaco-917.mdx

```diff
@@ -0,0 +1,91 @@
+---
+id: usaco-917
+source: USACO Bronze February 2019
+title: Measuring Traffic
+author: Brad Ma
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_traffic_bronze_feb19.html)
+
+## Implementation
+**Time Complexity** $\mathcal{O}(N)$
+
+<LanguageSection>
+
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class MeasuringTraffic {
+	static final int infinity = 999999999;
```

Comment:
Integer.MAX_VALUE is better for this
```suggestion
	static final int infinity = Integer.MAX_VALUE;
```

Reply:
or better yet do away with the constant entirely and just use `Integer.MAX_VALUE`

Reply:
wait nvm
overflow is a possibility

Reply:
that works too, it won't overflow because the bounds will be updated by the time you add/subtract stuff

Reply:
![image](https://user-images.githubusercontent.com/55369003/126888519-4ff1cb25-5720-462d-ac7a-2f93044bf10a.png)
no i think it will

Reply:
nah, when you do Math.max, low will instantly update to whatever start[i[ is. Same with high.

Reply:
bold of you to assume that the max will be executed first

Reply:
wait also @brad182 can you convert these to else ifs

Reply:
whoops you're right

Reply:
yep he's right and i tried it too. the second if executes first in the sample input

Reply:
> wait also @brad182 can you convert these to else ifs

also add a comment explaining why integer.max_value wouldn't work